### ISSUE TITLE
Scrolling the waveform with the horizontal scrollbar continues after the left mouse button is released

### ISSUE THREAD

**Issue by petersampsonaudacity** [audacity/audacity#6342]
### Bug description
Scrolling with the horizontal scrollbar continues after the left mouse button is released and the cursor remains in the scroll bar.

This is s regression on 3.4.2 and earlier

### Steps to reproduce
1. import some audio
2. zoom out several times to facilitate horizontal scrolling
3. Click and release in the gray space in the horizontal scroll bar to the right of the dark gray rectangle (position indicator) and immediately release the mouse button.
4. Observe: waveform advances by amount of time shown on the screen - EXPECTED BEHAVIOR
5.  Click and release in the gray space in the horizontal scroll bar to the right of the dark gray rectangle (position indicator) and hold mouse button down.
6.  Observe: scrolling carries on continuously - EXPECTED BEHAVIOR
7. Release the mouse button
8. Observe: scrolling continues  -  REGRESSION BEHAVIOR

_If you move the cursor out of the scroll bar (without the mouse button being held down) then scrolling stops but if you then move the cursor back into the scroll bar to the right of the gray rectangle the scrolling continues merrily along gain._

### Expected behavior
Scrolling should stop at step 7 when the LMB is released

### Actual behavior
Scrolling does NOT stop at step 7 when the LMB is released

### Audacity Version
3.5.0/3.5.1

### Operating system
Windows 10

### Additional context
Originally raised on the Forum:  https://forum.audacityteam.org/t/issue-with-latest-update/104665

**Comment by Lepereafelix** [comment]
Agree, this big is unfortunately annoying

**Comment by petersampsonaudacity** [comment]
@LWinterberg @Tantacrul @dozzzzer 

On the Forum user @AndyMusik sees this new behavior as beneficial, a useful new feature and **_NOT_** a bug to be fixed.  

He wrote in https://forum.audacityteam.org/t/issue-with-latest-update/104665:

>Transport Continuous Scrolling…
>If ticked…then will just scroll horizontal small increments when clicked on horiz scroll bar…
If not ticked … then will scroll the screen width in time each click on the horizontal scroll bar at bottom…
click anywhere to right of slider curser in bar for screem moves to left, click anywhere to left of curser for screen moves screen width to right.
>
>Long click hold puts the scrolling into continuous scroll while mouse is over the bar,… move mouse off the bar to stop this continuous scroll and move mouse back over bar to start it again no clicks…
OR click twice over the bar to go back to click for individual screen width scroll again.
I hope I am on the same subject…its a built in feature…???

and further he wrote:

> I think it is a good feature… with several options
With continuous scrolling ticked you can incremental scroll a few seconds with each click.
With continuous scrolling un-ticked you can scroll a screen width with each click.
If you dont want to wear out your finger or the mouse button…do a long click and continuous scrolling will start controlled by the mouse hover position… if hovering over the bar scrolling starts to roll and hover away from the bar and the scrolling roll stops, hover back to bar and it starts again…no clicks involved…??
>
>I think I am wrong about the need to Tick Continuous Monitoring in the Transport menu…its not required.
Click the Right arrow at end of horiz scroll bar to get incremental few seconds scroll…
Then as described above click on the space on the bar to get screen width scroll per click
and hold click to get continuous rolling scroll controlled by hovering the mouse position
>
>The manual that is in the menu of 3.5.1 does not describe Continuous scrolling as it is in the transport menu of 3.5.1… so menu is out of date or manual is outof date…??

**Comment by KoenTanghe** [comment]
I also just ran into this bug today (initially thinking my mouse was behaving erratically).
Setup: Audacity 3.5.1, Windows 10, using a Logitech MX Master 3 mouse (but that probably doesn't matter for this bug).

@LWinterberg @Tantacrul 
As this doesn't seem to be part of a Project yet, I would really suggest to categorize this as a major user experience bug that should be fixed ASAP, because scrolling is such a basic action for an audio editor (or DAW). Just try this once, and you'll immediately think "what is going on here?".

Another suggestion:
It is very weird that such a basic thing slipped through quality control (did no one actually open an audio file and scroll during testing? not even once?) Maybe this should be added to the checklist for QA testers?

Somewhat related:
If someone is going to be handling the code around horizontal scrolling to fix this bug, he/she/they might as well have a look at this longstanding related bug too: https://github.com/audacity/audacity/issues/4786

**Comment by LWinterberg** [comment]
I can reproduce this in around 1 in 20 clicks in 3.6.

**Comment by dozzzzer** [comment]
I can reproduce this at about the same rate as @LWinterberg on Win10 / latest master

**Comment by KoenTanghe** [comment]
For me, it's not once in a while: it happens always (3.5.1).
I attached a screen recording: 

https://github.com/audacity/audacity/assets/872038/94bfd8dc-ae66-4954-bbaf-2fccc0a9dc46

**Comment by KoenTanghe** [comment]
Small update for completeness' sake, as I had updated recently: Audacity 3.6.1 also has this same bug, and it also happens every time here for me in this version (on Windows 10 latest as of today).

**Comment by KoenT-SS** [comment]
This UX bug with the sticky behavior of the scroll bar is still present in the latest Audacity 3.7.1, and currently shows up like this for me (it doesn't matter if "Transport -> Continuous scrolling" is ticked or not):

1. zoom in several levels
2. left-click in the scrollbar somewhere at the right next to the dark gray scroll handle and release the mouse button

Observed:
Scrolling continues to the right until the scroll handle reaches the mouse location (when it's still in the scroll bar), even though you already released the mouse button. The scrolling state essentially gets stuck, and the only way to get rid of this erratic sticky behavior is by clicking on the scroll bar handle.

Expected:
Scrolling should only happen until you release the mouse button (and it should result in just 1 single zoom view width "scroll step" if you release the mouse button immediately), just like any other program.

Could this please get some attention so it finally ends up in one of the AU3 or AU4 boards?
This is such a basic user interface action, and really hinders the user experience when you want to inspect something / navigate in the sound (and especially when you try to click just once in the scroll bar to make it scroll 1 single zoom view width).

**Comment by BartyMae** [comment]
Yeah, I just googled "what is wrong with Audacity's crappy scroll bar?" and found this thread. Even when you do shift-click scrolls (which makes it go to the exact spot that you click), it continues to scroll away from where you click afterwards, which is utterly maddening.

**Comment by BartyMae** [comment]
Yeah, I just googled "what is wrong with Audacity's scroll bar?" and found this thread. Even when you do shift-click scrolls (which makes it go directly to the exact spot that you click), it continues to scroll away from where you click afterwards, which is utterly maddening. It happens literally every time I click on the scroll bar.

**Comment by KoenT-SS** [comment]
Great to see this being worked on! Thanks!
If you have a link to a binary for Windows that contains this fix I could also test it here a bit, if that would be useful.
Otherwise I'll just wait for the next release ;-)

**Comment by kryksyh** [comment]
@KoenTanghe here you go: https://github.com/audacity/audacity/actions/runs/12950263272

**Comment by KoenT-SS** [comment]
@kryksyh Thanks!
Just tested with audacity-win-3.7.1-alpha-20250124+223fbd9-x64-msvc2022.zip on Windows 10 (latest as of today) and I can confirm that this now works as expected (including the shift-clicking I saw someone else mention before).
Thanks for the fix!
